### PR TITLE
Adds a flare gun

### PR DESCRIPTION
**Pull request by Smiruk**
## About The Pull Request

Adds a flare gun, flare shells, puts them on tool design disk, reworks lighting to make lights colorable(done by mighty SirRichardFrancis), allows projectiles to emit light(also done by mighty SirRichardFrancis).
Also changes the light that carbons emit when on fire to be coloured red(because it makes sense)
Makes flare gun be wearable in most belts, spawns flare gun with some flare shells in expedition lockers, miners equipment lockers, at the trash beacon, in IH office and a small chance to spawn in emergency toolbox.

## Why It's Good For The Game

A flare gun is a somewhat iconic last-resort weapon, appearing in many movies and videogames.
It should fill the niche of a trash weapon any vagabond can find and use, makes for a good getaway weapon because it has low damage and occurse few debuffs on target.
Also adds a mechanic of lighting up a single direction - could make for some good in-game scenarios, such as shooting a shell in a dark area of maintenance to reveal something horrific from a safe distance.

## Changelog
:cl:
add: Added a flare gun. "How do you like your flesh cooked?"
fix: Fixed some Guild's Random Wave Radio typos
/:cl:

### CODE REVIEW

**Comment by SirRichardFrancis** [comment]
image
You're deleting Eris map, my dude. Please don't.

**Comment by Smiruk** [comment]
Lmao, how. Gotta fix this tomorrow

**Comment by Smiruk** [comment]
fixed it

File: code/datums/autolathe/ammo.dm [hunk 1]
```diff
@@ -358,3 +358,6 @@
 	name = "chemical darts magazine"
 	build_path = /obj/item/ammo_magazine/chemdart
 
+/datum/design/autolathe/ammo/flare_shell
+	name = "flare shell"
+	build_path = /obj/item/ammo_casing/flare
```
Comment (TheShown911) [inline code/datums/autolathe/ammo.dm:363]: No newline(there should be an empty line before the end of the file).

File: code/datums/autolathe/guns.dm [hunk 2]
```diff
@@ -291,3 +291,7 @@
 /datum/design/autolathe/gun/dart_gun
 	name = "Z-H P Artemis Dartgun"
 	build_path = /obj/item/gun/projectile/dartgun
+
+/datum/design/autolathe/gun/flare_gun
+	name = "flare gun"
+	build_path = /obj/item/gun/projectile/flare_gun
```
Comment (TheShown911) [inline code/datums/autolathe/guns.dm:297]: No newline.

File: code/game/objects/effects/effect_system.dm [hunk 3]
```diff
@@ -268,29 +268,34 @@ steam.start() -- spawns the effect
...
 	icon_state = "sparks"
 
-/obj/effect/effect/smoke/illumination/New(var/newloc, var/brightness=15, var/lifetime=10)
-	time_to_live=lifetime
-	..()
-	set_light(brightness)
+/obj/effect/effect/smoke/illumination/New(var/newloc, var/brightness=15, var/lifetime=10, var/color=COLOR_WHITE)
+    time_to_live=lifetime
+    ..()
+    set_light(brightness, 1, color)
```
Comment (TheShown911) [inline code/game/objects/effects/effect_system.dm:298]: Are these tabbed correctly?

File: code/modules/projectiles/ammunition/bullets.dm [hunk 4]
```diff
@@ -464,3 +464,17 @@
...
+	caliber = CAL_FLARE
+	icon_state = "f-shell"
+	spent_icon = "f-shell-spent"
+	projectile_type = /obj/item/projectile/bullet/flare
+	matter = list(MATERIAL_PLASTIC = 1)
+	matter_reagents = list("phosphorus" = 3)
+	rarity_value = 8
+
+/obj/item/ammo_casing/flare
+	amount = 3
```
Comment (TheShown911) [inline code/modules/projectiles/ammunition/bullets.dm:480]: No newline.

File: code/modules/projectiles/guns/projectile/flaregun.dm [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+/obj/item/gun/projectile/flare_gun/on_update_icon()
+	..()
+
+	var/iconstring = initial(icon_state)
+
+	if (bolt_open)
+		iconstring += "_open"
+
+	icon_state = iconstring
```
Comment (TheShown911) [inline code/modules/projectiles/guns/projectile/flaregun.dm:31]: Should use `SetIconState()`

File: code/modules/projectiles/projectile/special.dm [hunk 6]
```diff
@@ -169,3 +169,47 @@
...
+	//blind adjacent people with enhanced vision
+	for (var/mob/living/carbon/M in viewers(T, flash_range))
+		if(M.eyecheck() < FLASH_PROTECTION_NONE)
+			if (M.HUDtech.Find("flash"))
+				FLICK("e_flash", M.HUDtech["flash"])
+
+	src.visible_message(SPAN_WARNING("\The [src] explodes in a bright light!"))
+	new /obj/effect/decal/cleanable/ash(src.loc)
+	playsound(src, 'sound/effects/flare.ogg', 100, 1)
+	new /obj/effect/effect/smoke/illumination(T, brightness=max(flash_range*3, brightness), lifetime=light_duration, color=COLOR_RED)
```
Comment (TheShown911) [inline code/modules/projectiles/projectile/special.dm:215]: No newline.

**Comment by Smiruk** [comment]
should be all fixed now

**Review by TheShown911 — APPROVED** [review]
Make sure to update the PR changelog when adding new stuff to the PR.

**Pull request merged**